lightning-invoice/src/utils.rs

```diff
@@ -289,19 +288,18 @@ where
 /// Draw items iteratively from multiple nested vectors.  The items are retrieved by index and
 /// rotates through the vectors - first the zero index then the first index then second index, etc.
 fn rotate_nested_vectors<T, I: Iterator<Item = T>>(mut vecs: Vec<I>) -> impl Iterator<Item = T> {
-	let mut idx = 0;
+	let mut iterations = 0;
 
 	core::iter::from_fn(move || {
 		let mut exhausted_vectors = 0;
```

Comment:
While we are cleaning up naming, I believe the use of vecs/vectors in here should really be iterators. Likewise in the documentation.